api/prometheus/v1/api.go

```diff
@@ -830,6 +861,13 @@ func WithTimeout(timeout time.Duration) Option {
 	}
 }
 
+func WithQueryStats(s *QueryStats, perStep bool) Option {
```

Comment:
This is bit unexpected as a start what will return value via variadic option (: So let's add a comment at least.

But I would be curious if we can actually change `model.Value` with some interface that is like model.Value but also returns stats 🤔 Will be much nicer for users.

And this package is in experimental stage so we can change API btw.

Reply:
> And this package is in experimental stage so we can change API btw.

Brilliant, I'll try to come up with an API change that makes it easier to consume this.